benchmark/benchmark.cpp

```diff
@@ -0,0 +1,358 @@
+
+// The benchmark arguments specified in Args() are, in order:
+// benchmark arguments, allocator arguments, size generator arguments.
+// The exact meaning of each argument depends on the benchmark, allocator, and size components used.
+// Refer to the 'name()' function in each component to find detailed descriptions of these arguments.
+UMF_BENCHMARK_REGISTER_F(alloc_benchmark, glibc_fix)
+    ->Args({10000, 0, 4096})
+    ->Args({10000, 100000, 4096})
+    ->Threads(4)
+    ->Threads(1);
+
+UMF_BENCHMARK_TEMPLATE_DEFINE(alloc_benchmark, glibc_uniform,
+                              uniform_alloc_size, glibc_malloc);
+UMF_BENCHMARK_REGISTER_F(alloc_benchmark, glibc_uniform)
+    ->Args({10000, 0, 8, 64 * 1024, 8})
+    ->Threads(4)
+    ->Threads(1);
+
+UMF_BENCHMARK_TEMPLATE_DEFINE(alloc_benchmark, os_provider, fixed_alloc_size,
+                              provider_allocator<os_provider>);
+UMF_BENCHMARK_REGISTER_F(alloc_benchmark, os_provider)
+    ->Args({10000, 0, 4096})
+    ->Args({10000, 100000, 4096})
+    ->Threads(4)
+    ->Threads(1);
+
+UMF_BENCHMARK_TEMPLATE_DEFINE(alloc_benchmark, proxy_pool, fixed_alloc_size,
+                              pool_allocator<proxy_pool<os_provider>>);
+
+UMF_BENCHMARK_REGISTER_F(alloc_benchmark, proxy_pool)
+    ->Args({1000, 0, 4096})
+    ->Args({1000, 100000, 4096})
+    ->Threads(4)
+    ->Threads(1);
+
+#ifdef UMF_BUILD_LIBUMF_POOL_DISJOINT
+UMF_BENCHMARK_TEMPLATE_DEFINE(alloc_benchmark, disjoint_pool_fix,
+                              fixed_alloc_size,
+                              pool_allocator<disjoint_pool<os_provider>>);
+UMF_BENCHMARK_REGISTER_F(alloc_benchmark, disjoint_pool_fix)
+    ->Args({10000, 0, 4096})
+    ->Args({10000, 100000, 4096})
+    ->Threads(4)
+    ->Threads(1);
+
+// TODO: debug why this crashes
+/*UMF_BENCHMARK_TEMPLATE_DEFINE(alloc_benchmark, disjoint_pool_uniform,
+                              uniform_alloc_size,
+                              pool_allocator<disjoint_pool<os_provider>>);
+UMF_BENCHMARK_REGISTER_F(alloc_benchmark, disjoint_pool_uniform)
+    ->Args({10000, 0, 8, 64 * 1024, 8})
+    ->Threads(4)
+    ->Threads(1);*/
+
+#endif
+
+#ifdef UMF_BUILD_LIBUMF_POOL_JEMALLOC
+UMF_BENCHMARK_TEMPLATE_DEFINE(alloc_benchmark, jemalloc_pool_fix,
+                              fixed_alloc_size,
+                              pool_allocator<jemalloc_pool<os_provider>>);
+UMF_BENCHMARK_REGISTER_F(alloc_benchmark, jemalloc_pool_fix)
+    ->Args({10000, 0, 4096})
+    ->Args({10000, 100000, 4096})
+    ->Threads(4)
+    ->Threads(1);
+
+UMF_BENCHMARK_TEMPLATE_DEFINE(alloc_benchmark, jemalloc_pool_uniform,
+                              uniform_alloc_size,
+                              pool_allocator<jemalloc_pool<os_provider>>);
+UMF_BENCHMARK_REGISTER_F(alloc_benchmark, jemalloc_pool_uniform)
+    ->Args({10000, 0, 8, 64 * 1024, 8})
+    ->Threads(4)
+    ->Threads(1);
+
+#endif
+
+UMF_BENCHMARK_TEMPLATE_DEFINE(alloc_benchmark, scalable_pool_fix,
```

Comment:
shouldn't we also have `ifdef` for scalable pool enabled?

Reply:
done